lab_1_keywords_tfidf/main.py

```diff
@@ -2,158 +2,256 @@
-
-    In case of corrupt input arguments, None is returned
+    Checks that type of 'list1' is list and verifies the contents of 'list1' with the type that the user specifies
     """
-    pass
+    if isinstance(list1, list):
+        if not list1 and not empty:
+            return False
+        for index in list1:
+            if not isinstance(index, type1):
+                return False
+        return True
+    return False
 
 
-def remove_stop_words(tokens: list[str], stop_words: list[str]) -> Optional[list[str]]:
+def correct_dict(dictionary: dict, type1: Any, type2: Any, empty: bool) -> bool:
     """
-    Excludes stop words from the token sequence
-
-    Parameters:
-    tokens (List[str]): Original token sequence
-    stop_words (List[str]: Tokens to exclude
-
-    Returns:
-    List[str]: Token sequence that does not include stop words
-
-    In case of corrupt input arguments, None is returned
+    Checks that type of 'dictionary' is dict and verifies the contents of 'dictionary' with the type that
+    the user specifies
     """
-    pass
+    if isinstance(dictionary, dict):
+        if not dictionary and not empty:
+            return False
+        for key, value in dictionary.items():
+            if not isinstance(key, type1) or not isinstance(value, (int, type2)) or isinstance(value, bool):
+                return False
+        return True
+    return False
 
 
-def calculate_frequencies(tokens: list[str]) -> Optional[dict[str, int]]:
+def clean_and_tokenize(text: str) -> Optional[list[str]]:
     """
-    Composes a frequency dictionary from the token sequence
+        Removes punctuation, casts to lowercase, splits into tokens
+        Parameters:
+        text (str): Original text
+        Returns:
+        list[str]: A sequence of lowercase tokens with no punctuation
+        In case of corrupt input arguments, None is returned
+        """
+    if not isinstance(text, str):
+        return None
+    lowered_text = text.lower()
+    for punctuation_mark in string.punctuation:
+        if punctuation_mark in lowered_text:
+            lowered_text = lowered_text.replace(punctuation_mark, '')
+    cleaned_text = lowered_text.strip().split()
+    return cleaned_text
 
-    Parameters:
-    tokens (List[str]): Token sequence to count frequencies for
 
-    Returns:
-    Dict: {token: number of occurrences in the token sequence} dictionary
-
-    In case of corrupt input arguments, None is returned
+def remove_stop_words(tokens: list[str], stop_words: list[str]) -> Optional[list[str]]:
     """
-    pass
+        Excludes stop words from the token sequence
+        Parameters:
+        tokens (List[str]): Original token sequence
+        stop_words (List[str]: Tokens to exclude
+        Returns:
+        List[str]: Token sequence that does not include stop words
+        In case of corrupt input arguments, None is returned
+        """
+    if correct_list(tokens, str, False) and correct_list(stop_words, str, True):
```

Comment:
reverse the condition and return `None` as soon as possible

Reply:
Done